pkg/kv/fs.go

```diff
@@ -0,0 +1,67 @@
+package kv
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"sync"
+
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+type FSStore struct {
+	lock   *sync.RWMutex
+	fsPath string
+}
+
+func NewFSStore(logger *zap.Logger, fsPath string) *FSStore {
+	return &FSStore{
+		lock:   new(sync.RWMutex),
+		fsPath: fsPath,
+	}
+}
+
+func (s *FSStore) Start(ctx context.Context, g *errgroup.Group) error {
+	return nil
+}
+func (s *FSStore) Get(ctx context.Context, ns Namespace, key string) (string, error) {
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+
+	path := fmt.Sprintf("%s/%s/%s", s.fsPath, ns, key)
+	b, err := os.ReadFile(path)
+
+	if errors.Is(err, os.ErrNotExist) {
+		s.Touch(path)
+		return "", nil
+	}
+	if err != nil {
+		return "", err
+	}
+	return string(b), nil
+}
+
+func (s *FSStore) Set(ctx context.Context, ns Namespace, key string, value string) error {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	path := fmt.Sprintf("%s/%s/%s", s.fsPath, ns, key)
+	s.Touch(path)
+	return os.WriteFile(path, []byte(value), 0644)
+}
+
+func (s *FSStore) Touch(fpath string) error {
+	baseDir := path.Dir(fpath)
+	if err := os.MkdirAll(baseDir, 0755); err != nil {
+		return err
+	}
+
+	if file, err := os.OpenFile(fpath, os.O_RDONLY|os.O_CREATE, 0644); err != nil {
+		return err
+	} else {
+		return file.Close()
+	}
```

Comment:
1. If all branches of the if-elseif-else are returning something, usually we don't write the else case because it will create too much indentation and make worse readability.

Reply:
2. Golang people like multiple ifs and check err one by one

```go
file, err := os.OpenFile(...)
if err != nil {
  return err
}

err = file.Close()
if err != nil {
  return err
}

return nil
```

Reply:
So strange, why not just `return file.Close()` 🤔 I will apply this change to `os.WriteFile` call in `Set` also